hanagotchi-app/src/screens/CompleteLoginScreen.tsx

```diff
@@ -0,0 +1,148 @@
+    const hanagotchiApi = useHanagotchiApi();
+    const { signOut , completeSignIn } = useAuth();
+    const { uploadImage } = useFirebase();
+    const { get } = useLocalStorage();
+    const { location, requestLocation, revokeLocation, changeLocation } = useLocation();
+
+    const [user, setUser] = useState<UserData>({ id: -1, email: '' } as UserData);
+
+    const genders = [
+        { key: 1, value: "HOMBRE" },
+        { key: 2, value: "MUJER" },
+        { key: 3, value: "OTRO" },
+    ]
+    const fetchUser = async () => {
+        const user_id = await get("user_id");
+        if (!user_id) {
+            console.log("No user_id found");
+            return;
+        }
+        const { message: userFetched }: LoginResponse = await hanagotchiApi.getUser(Number(user_id));
+        // setDateOfBirth(new Date(user.dateOfBirth ?? ''));s
+        setUser(userFetched as UserData);
+    }
+
+
+    const handleBirthDay = async (date: Date) => {
+        setUser({ ...user, dateOfBirth: date.toDateString() } as UserData);
+    }
+    const handleName = (name: string) => {
+        setUser({ ...user, name: name } as UserData);
+    }
+    const handleGender = (val: string) => {
+        setUser({ ...user, gender: val } as UserData);
+    }
+
+    const handleRegionChange = (region: Region, details: Details) => {
+        changeLocation(region.latitude, region.longitude, region.latitudeDelta, region.longitudeDelta);
+    }
+
+    const handleUploadPhoto = async () => {
+        const permissionResult = await ImagePicker.requestMediaLibraryPermissionsAsync();
+        if (permissionResult.granted === false) {
+            alert('¡Se requiere permiso para acceder a la galería de imágenes!');
+            return;
+        }
+        const result = await ImagePicker.launchImageLibraryAsync({
+            mediaTypes: ImagePicker.MediaTypeOptions.Images,
+            allowsEditing: true,
+            aspect: [4, 4],
+            quality: 1,
+        });
+        if (!result.canceled) {
+            setUser({ ...user, photo: result.assets[0].uri } as UserData);
+        }
+    };
+
+    useEffect(() => {
+        const fetchData = async () => {
+            await requestLocation();
+            await fetchUser();
+        };
+
+        fetchData();
+    }, []);
+
+    useEffect(() => {
+        if (location !== FIUBA_REGION && user.location !== location) {
+            setUser({ ...user, location: location });
+        }
+    }, [location]);
+
+    useEffect(
+        () =>
+            // https://reactnavigation.org/docs/preventing-going-back/
+            navigation.addListener('beforeRemove', async (e) => {
+                console.log("signOut start");
+                await signOut();
+                console.log("signOut success");
+                revokeLocation();
+                navigation.navigate("Login");
+            }),
+        [navigation]
+    );
+
+    const handleSignOut = async () => {
```

Comment:
Por que la funcion se llama handleSignOut si se esta usando para terminar el Sign In?